pkg/webui/console/views/gateway-general-settings/basic-settings-form/index.js

```diff
@@ -0,0 +1,200 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+
+import SubmitButton from '@ttn-lw/components/submit-button'
+import SubmitBar from '@ttn-lw/components/submit-bar'
+import Form from '@ttn-lw/components/form'
+import Input from '@ttn-lw/components/input'
+import Checkbox from '@ttn-lw/components/checkbox'
+import KeyValueMap from '@ttn-lw/components/key-value-map'
+import ModalButton from '@ttn-lw/components/button/modal-button'
+
+import Require from '@console/lib/components/require'
+
+import diff from '@ttn-lw/lib/diff'
+import PropTypes from '@ttn-lw/lib/prop-types'
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+
+import { mapFormValueToAttributes, mapAttributesToFormValue } from '@console/lib/attributes'
+
+import m from '../messages'
+
+import validationSchema from './validation-schema'
+
+const BasicSettingsForm = React.memo(props => {
+  const {
+    gateway,
+    onSubmit,
+    onSubmitSuccess,
+    onDelete,
+    onDeleteFailure,
+    onDeleteSuccess,
+    mayDeleteGateway,
+  } = props
+
+  const [error, setError] = React.useState('')
```

Comment:
```suggestion
  const [error, setError] = React.useState(undefiend)
```

Nit, but it rellay doesn't make sense that we always use an empty string for this.